src/components/type/type.mapper.ts

```diff
@@ -0,0 +1,22 @@
+import { Type } from './type.model';
+import { ItemTemplate } from '@core/game_master/gameMaster';
+import { Util } from '@util';
+
+export class TypeMapper {
+    public static Map(rawType: ItemTemplate, typeIds: Map<number, string>): Type {
+        let type: Type = new Type();
+
+        type.id = rawType.templateId
+        type.name = Util.SnakeCase2HumanReadable(rawType.templateId.replace('POKEMON_TYPE_', ''));
+
+        type.damage = [];
+        for (let i = 0; i < rawType.typeEffective.attackScalar.length; i++) {
```

Comment:
Can you use 

```JavaScript
rawType.typeEffective.attackScalar.forEach((attackScalar, index) => { .. });
```

 instead of `for` ?

Reply:
With POKEMON_TYPES array defined in app.settings.ts, it was much easier to just use a map to do the job.